src/actions.rs

```diff
@@ -478,14 +478,10 @@ impl ActionHandler {
         let mut buf = Vec::<u8>::new();
         match format_input(input, &config, Some(&mut buf)) {
             Ok((summary, ..)) => {
-                // It's workaround because currently rustfmt returns ok even if there are parsing errors.
-                // If there are parsing errors the buffer is empty.
-                // Without this condition a text document is erased.
-                if summary.has_parsing_errors() {
-                    debug!("reformat: format_input failed: has parsing errors");
-
-                    out.failure(id, "Reformat failed to complete successfully")
-                } else {
+                // format_input returns Ok even if there are any errors, i.e., parsing errors.
+                // If there are any errors the buffer is empty.
```

Comment:
That doesn't seem like a good comment to leave - the first part explains why you are checking this, and this second bit could change if Rustfmt changes and then the comment would be misleading.